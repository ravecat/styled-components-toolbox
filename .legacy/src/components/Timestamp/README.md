## Timestamp

A localized timestamp like _Oct 7 2018 10:25_

### API

| Property  | Description                                             | Type                 | Default |
| --------- | ------------------------------------------------------- | -------------------- | ------- |
| _mask_    | set fields included in the timestamp                    | string               | `bDYt`  |
| _\*value_ | the date value as an ISO8601 string or a `Date` object. | Date, ISO8601 string | -       |

Also can use `mask` attribute for set format timestamp (example `2011-10-09T14:48:59`)

| Symbol | Description              | Example    |
| ------ | ------------------------ | ---------- |
| `Y`    | year                     | _2011_     |
| `M`    | month (decimal value)    | _10_       |
| `D`    | day (decimal value)      | _09_       |
| `m`    | minutes                  | _48_       |
| `h`    | hours                    | _14_       |
| `s`    | seconds                  | _59_       |
| `a`    | day of week (short form) | _Tue_      |
| `A`    | day of week (long form)  | _Tuesday_  |
| `b`    | month (short form)       | _2011_     |
| `B`    | month (long form)        | _2011_     |
| `t`    | time                     | _14:48:69_ |

Also you can use different symbols for decorate your timestamp (`|/\*:.,`)

### Example

Below is the result of using different masks for `2011-10-09T14:48:59`

| Mask    | Result                   |
| ------- | ------------------------ |
| `bDYt`  | _Oct 09 2011 14:48:59_   |
| `ADBY`  | _Sunday 09 October 2011_ |
| `DBh:m` | _09 October 14 : 48_     |
| `Y/M/D` | _2011 / 10 / 09_         |
| `DMYt`  | _09 10 2011 14:48:59_    |
