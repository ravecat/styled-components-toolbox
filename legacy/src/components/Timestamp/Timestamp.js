import React from "react";
import styled from "styled-components";

import { theme } from "common/theme";
import BaseComponent from "components/BaseComponent";

const Timestamp = ({ date, className }) => (
  <Wrapper className={className}>{date}</Wrapper>
);

export default Timestamp;

const Wrapper = styled(BaseComponent).attrs({
  as: "span"
})`
  color: ${theme.primaryTextColor};
`;
