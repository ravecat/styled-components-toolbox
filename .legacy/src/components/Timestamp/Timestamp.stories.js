import React from "react";
import { text, select } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";

import Timestamp from "./index";

const DEFAULT = "2018-10-08T18:31:42";
const OPTIONS = [
  "2018-10-08T18:31:42",
  "2017-09-08",
  "2016-08-08T18:31:42.201",
  "--06-08",
  "2015-07-08T21:48:51Z",
  "2014-06-08T21:48:51+00:00"
];

storiesOf("Timestamp", module).add("Default", () => (
  <Timestamp
    mask={text("mask", "bDYt")}
    value={select("value", OPTIONS, DEFAULT)}
  />
));
