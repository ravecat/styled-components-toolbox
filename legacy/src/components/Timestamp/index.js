import withDate from "./services/withDate";
import Timestamp from "./Timestamp";

export default withDate(Timestamp);
