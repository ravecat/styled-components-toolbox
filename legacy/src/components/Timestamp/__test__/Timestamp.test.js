import React from "react";
import renderer from "react-test-renderer";

import Timestamp from "../index";

it("<Timestamp> renders correctly with Date object", () => {
  const date = new Date(2011, 9, 9, 14, 48, 59);
  const tree = renderer.create(<Timestamp value={date} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("<Timestamp> renders correctly with ISO8061 string", () => {
  const tree = renderer
    .create(<Timestamp value={"2011-10-09T14:48:59"} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("<Timestamp> renders correctly with ISO8061 string and 'YBt' mask", () => {
  const tree = renderer
    .create(<Timestamp mask={"YBt"} value={"2011-10-09T14:48:59"} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("<Timestamp> renders correctly with Date object and 'YBt' mask", () => {
  const date = new Date(2011, 9, 9, 14, 48, 59);
  const tree = renderer
    .create(<Timestamp mask={"YBt"} value={date} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
