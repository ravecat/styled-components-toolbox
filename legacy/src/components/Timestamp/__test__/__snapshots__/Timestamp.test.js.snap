// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Timestamp> renders correctly with Date object 1`] = `
<span
  className="sc-bdVaJa sc-bwzfXH gGFlHM"
>
  Oct 09 2011 14:48:59
</span>
`;

exports[`<Timestamp> renders correctly with Date object and 'YBt' mask 1`] = `
<span
  className="sc-bdVaJa sc-bwzfXH gGFlHM"
>
  2011 October 14:48:59
</span>
`;

exports[`<Timestamp> renders correctly with ISO8061 string 1`] = `
<span
  className="sc-bdVaJa sc-bwzfXH gGFlHM"
>
  Oct 09 2011 14:48:59
</span>
`;

exports[`<Timestamp> renders correctly with ISO8061 string and 'YBt' mask 1`] = `
<span
  className="sc-bdVaJa sc-bwzfXH gGFlHM"
>
  2011 October 14:48:59
</span>
`;
